src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -618,6 +618,7 @@ void primitive_inst::realloc_if_needed() {
                     _max_output_layout_count[j] = 0;
                 }
             } else {
+                _outputs[0] = variable.get_memory();
```

Comment:
I have some questions about this PR:
* This is setting `_outputs[0]` on `realloc_if_needed`. In case realloc does not happen, are we using proper buffer for `_outputs[0]`? I'm afraid this should be set before this function.
* Does `shape_predictor` work correctly for multi-ireq scenario? From the code, it seems like that the shape_predictor is placed within the network and it may have issue for multi-ireq scenario as it does not receive ireq information. We may need to take ireq information as an argument for that. Could you comment on it? @sshlyapn

Reply:
@isanghao I think there shouldn't be any problems with shape_predictor, since each ireq has unique instance of shape_predictor (which is passed to the network after the network is assigned to ireq):
https://github.com/openvinotoolkit/openvino/blob/0f149e39ed7dba9fae09fa6b56bb9f08f117aa9f/src/plugins/intel_gpu/include/intel_gpu/plugin/sync_infer_request.hpp#L89

Reply:
oh you already did the work ;) Thanks for confirmation.

Reply:
@sshlyapn  Memory of variable-state in two infer-requests are shared at this sample code.[minicpm_multi_inferreq.cpp](https://jira.devtools.intel.com/secure/attachment/4860298/minicpm_multi_inferreq.cpp)
results of query_state() for llmIR are changed when it calls llmIR2.infer().

Reply:
@sungeunk , thank you for the details. In that case, it seems we have to update the scales/zero points buffers as well